src/Symfony/Bundle/FrameworkBundle/Test/WebTestAssertions.php

```diff
@@ -0,0 +1,296 @@
+    {
+        self::checkClientAvailable();
+
+        $cookie = static::$client->getCookieJar()->get($name);
+
+        Assert::assertNull(
+            $cookie,
+            "Cookie [$name] was not expected to be set."
+        );
+    }
+
+    public static function assertCookieValue(string $name, $expectedValue, string $path = '/', string $domain = null): void
+    {
+        self::checkClientAvailable();
+
+        $cookie = static::$client->getCookieJar()->get($name, $path, $domain);
+
+        Assert::assertNotNull(
+            $cookie,
+            "Did not find expected cookie [$name]."
+        );
+        Assert::assertSame(
+            $expectedValue,
+            $value = $cookie->getRawValue(),
+            "Cookie name [$name] with value [$expectedValue] does not match actual value [$value]."
+        );
+    }
+
+    public static function assertRawCookieValue(string $name, $expectedValue, string $path = '/', string $domain = null): void
+    {
+        self::checkClientAvailable();
+
+        $cookie = static::$client->getCookieJar()->get($name, $path, $domain);
+
+        Assert::assertNotNull(
+            $cookie,
+            "Did not find expected cookie [$name]."
+        );
+        Assert::assertSame(
+            $expectedValue,
+            $value = $cookie->getRawValue(),
+            "Cookie name [$name] with raw value [$expectedValue] does not match actual value [$value]."
+        );
+    }
+
+    public static function assertSelectorExists(string $selector): void
+    {
+        self::checkCrawlerAvailable();
+
+        $nodes = static::$client->getCrawler()->filter($selector);
+
+        Assert::assertGreaterThan(0, $nodes->count(), "Selector [$selector] does not resolve to any node.");
+    }
+
+    public static function assertSelectorNotExists(string $selector): void
+    {
+        self::checkCrawlerAvailable();
+
+        $nodes = static::$client->getCrawler()->filter($selector);
+
+        Assert::assertEquals(0, $count = $nodes->count(), "Selector [$selector] resolves to [$count] nodes where it expected 0.");
+    }
+
+    public static function assertSelectorContainsText(string $selector, string $text): void
+    {
+        self::checkCrawlerAvailable();
+
+        $nodes = static::$client->getCrawler()->filter($selector);
+
+        Assert::assertGreaterThan(0, $nodes->count(), "Selector [$selector] does not resolve to any node.");
+
+        Assert::assertContains($text, $nodes->text(), "Selector [$selector] does not contain text [$text].");
+    }
+
+    public static function assertSelectorNotContainsText(string $selector, string $text): void
+    {
+        self::checkCrawlerAvailable();
+
+        $nodes = static::$client->getCrawler()->filter($selector);
+
+        Assert::assertGreaterThan(0, $nodes->count(), "Selector [$selector] does not resolve to any node.");
+
+        Assert::assertContains($text, $nodes->text(), "Selector [$selector] does not contain text [$text].");
```

Comment:
Should this be

```php
Assert::assertNotContains($text, $nodes->text(), "Selector [$selector] contains text [$text].");
```

?

Reply:
Fixed, and I also reworded to `"Selector [$selector] was expected to not contain text [$text]."` instead, I think it is more explanatory 👍